### PR TITLE
[Marksmanship Hunter] Tier 20 2p and 4p 

### PR DESCRIPTION
**Pull request by Pewtro**
2p:
image
4p:
image

### CODE REVIEW

**Comment by Pewtro** [comment]
Welp, also the stuff from the other PR I made - so if that one is good to go, only this one needs to be accepted

File: src/Parser/MarksmanshipHunter/Modules/Items/Tier20_2p.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import Module from 'Parser/Core/Module';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import SpellLink from 'common/SpellLink';
+import { formatPercentage } from "../../../../common/format";
```
Comment (sMteX) [inline src/Parser/MarksmanshipHunter/Modules/Items/Tier20_2p.js]: Should be `'common/format'`.

File: src/Parser/MarksmanshipHunter/Modules/Items/Tier20_2p.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  }
+  on_byPlayer_damage(event) {
+    if(this.combatants.selected.hasBuff(SPELLS.HUNTER_MM_T20_2P_BONUS_BUFF.id) && (event.ability.guid === SPELLS.AIMED_SHOT.id)) {
+      this.buffedAimed += 1;
+      this.totalAimed += 1;
+    }
+    if(!this.combatants.selected.hasBuff(SPELLS.HUNTER_MM_T20_2P_BONUS_BUFF.id) && (event.ability.guid === SPELLS.AIMED_SHOT.id)) {
+      this.totalAimed +=1;
+    }
+
```
Comment (sMteX) [inline src/Parser/MarksmanshipHunter/Modules/Items/Tier20_2p.js]: Could be rewritten as 
```jsx
if (event.ability.guid !== SPELLS.AIMED_SHOT.id) {
   return;
}
if (this.combatants.selected.hasBuff(SPELLS.HUNTER_MM_T20_2P_BONUS_BUFF.id) {
   this.buffedAimed += 1;
}
this.totalAimed += 1;
```

File: src/Parser/MarksmanshipHunter/Modules/Items/Tier20_2p.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      id: `spell-${SPELLS.HUNTER_MM_T20_2P_BONUS_BUFF.id}`,
+      icon: <SpellIcon id={SPELLS.HUNTER_MM_T20_2P_BONUS_BUFF.id} />,
+      title: <SpellLink id={SPELLS.HUNTER_MM_T20_2P_BONUS_BUFF.id} />,
+      result: (
+<dfn data-tip={`Your utilization of tier 20 2 piece: <br/>
+Buffed aimed shots: ${this.buffedAimed}.<br/>
+Total aimed shots:  ${this.totalAimed}.<br/> `}>
+Buffed Aimed Shots:  {formatPercentage(this.buffedAimed / this.totalAimed)}%
+
+</dfn>
```
Comment (sMteX) [inline src/Parser/MarksmanshipHunter/Modules/Items/Tier20_2p.js]: Just a style issue - some indentation to the `<dfn>` would be nice, it's not very readable at the first sight what's part of the tooltip and what's not.

File: src/Parser/MarksmanshipHunter/Modules/Items/Tier20_4p.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import Module from 'Parser/Core/Module';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import SpellLink from 'common/SpellLink';
+import { formatPercentage } from "../../../../common/format";
```
Comment (sMteX) [inline src/Parser/MarksmanshipHunter/Modules/Items/Tier20_4p.js]: Should be `'common/format'`.

File: src/Parser/MarksmanshipHunter/Modules/Items/Tier20_4p.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    return uptime;
+  }
+
+  /*
+  // Irrelevant for now, hoping to find a way to register when a cast is only affected by tier at start of cast, because the way it functions is two-part.
+  on_byPlayer_begincast(event) {
+    if(this.combatants.selected.hasBuff(SPELLS.HUNTER_MM_T20_4P_BONUS_BUFF.id) && (event.ability.guid === SPELLS.AIMED_SHOT.id)) {
+      this.buffedCastAimed += 1;
+    }
+  }*/
```
Comment (sMteX) [inline src/Parser/MarksmanshipHunter/Modules/Items/Tier20_4p.js]: You should probably remove commented code like this (maybe make an issue on your fork repository as a reminder of this code so it doesn't get entirely lost?). Same with the commented line 15.

File: src/Parser/MarksmanshipHunter/Modules/Items/Tier20_2p.js (hunk 1 above)
Comment (sMteX) [inline src/Parser/MarksmanshipHunter/Modules/Items/Tier20_2p.js]: Should be `'common/format'`.

File: src/Parser/MarksmanshipHunter/Modules/Items/Tier20_2p.js (hunk 2 above)
Comment (sMteX) [inline src/Parser/MarksmanshipHunter/Modules/Items/Tier20_2p.js]: This should be rewritten as 
```jsx
if (event.ability.guid !== SPELLS.AIMED_SHOT.id) {
   return;
}
if (this.combatants.selected.hasBuff(SPELLS.HUNTER_MM_T20_2P_BONUS_BUFF.id, event.timestamp) {
   this.buffedAimed += 1;
}
this.totalAimed += 1;
```

File: src/Parser/MarksmanshipHunter/Modules/Items/Tier20_2p.js (hunk 3 above)
Comment (sMteX) [inline src/Parser/MarksmanshipHunter/Modules/Items/Tier20_2p.js]: Just a style issue - some indentation to the `<dfn>` would be nice - it's not readable at first sight what's part of the tooltip and what's not.

File: src/Parser/MarksmanshipHunter/Modules/Items/Tier20_4p.js (hunk 4 above)
Comment (sMteX) [inline src/Parser/MarksmanshipHunter/Modules/Items/Tier20_4p.js]: Should be `'common/format'`.

File: src/Parser/MarksmanshipHunter/Modules/Items/Tier20_4p.js (hunk 5 above)
Comment (sMteX) [inline src/Parser/MarksmanshipHunter/Modules/Items/Tier20_4p.js]: You should probably remove commented code like this (together with line 15). If you don't want to lose the code, maybe create an issue on your fork repository?

File: src/Parser/MarksmanshipHunter/Modules/Items/Tier20_4p.js [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+  on_byPlayer_cast(event) {
+    if(this.combatants.selected.hasBuff(SPELLS.HUNTER_MM_T20_4P_BONUS_BUFF.id) && (event.ability.guid === SPELLS.AIMED_SHOT.id)) {
+      this.buffedCastAndFocusAimed += 1;
+      this.totalAimed += 1;
+    }
+    if(!this.combatants.selected.hasBuff(SPELLS.HUNTER_MM_T20_4P_BONUS_BUFF.id) && (event.ability.guid === SPELLS.AIMED_SHOT.id)) {
+      this.totalAimed +=1;
+    }
+
```
Comment (sMteX) [inline src/Parser/MarksmanshipHunter/Modules/Items/Tier20_4p.js]: This should be rewritten as 
```jsx
if (event.ability.guid !== SPELLS.AIMED_SHOT.id) {
   return;
}
if (this.combatants.selected.hasBuff(SPELLS.HUNTER_MM_T20_4P_BONUS_BUFF.id, event.timestamp) {
   this.buffedCastAndFocusAimed += 1;
}
this.totalAimed += 1;
```

**Comment by Pewtro** [comment]
updated to meet requested changes

**Pull request merged**